src/units/abilities.cpp

```diff
@@ -1202,15 +1202,13 @@ std::pair<std::string, std::string> attack_type::damage_type() const
 		return {type(), ""};
 	}
 
-	std::vector<std::string> type_list = damage_type_list(abil_list, "replacement_type");
-	std::vector<std::string> added_type_list = damage_type_list(abil_list, "alternative_type");
-	std::string type_damage, sec_type_damage;
-	type_damage = !type_list.empty() ? type_list.front() : type();
-	sec_type_damage = !added_type_list.empty() ? added_type_list.front() : "";
-	if(!sec_type_damage.empty()){
-		sec_type_damage =  type_damage != sec_type_damage ? sec_type_damage: "";
+	std::optional<std::string> replacement_type = damage_type_list(abil_list, "replacement_type");
+	std::optional<std::string> alternative_type = damage_type_list(abil_list, "alternative_type");
+	std::string type_damage = replacement_type ? *replacement_type : type();
```

Comment:
Since this is an `optional` now you can use `replacement_type.value_or(type())`.

Reply:
He could, but in this case it doesn't seem to improve readability, IMO.

Reply:
I mean, if you're going to use optionals you should use their API.

Reply:
> Since this is an `optional` now you can use `replacement_type.value_or(type())`.

it's done

Reply:
For what it's worth, I agree with Vultraz on this one.